planner/optimize.go

```diff
@@ -193,10 +193,12 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 			bestPlanAmongHints = plan
 		}
 	}
-	// 1. If there is already a evolution task, we do not need to handle it again.
-	// 2. If the origin binding contain `read_from_storage` hint, we should ignore the evolve task.
-	// 3. If the best plan contain TiFlash hint, we should ignore the evolve task.
-	if sctx.GetSessionVars().EvolvePlanBaselines && binding == nil &&
+	// 1. If it is a select query.
+	// 2. If there is already a evolution task, we do not need to handle it again.
+	// 3. If the origin binding contain `read_from_storage` hint, we should ignore the evolve task.
+	// 4. If the best plan contain TiFlash hint, we should ignore the evolve task.
+	if _, ok := stmtNode.(*ast.SelectStmt); ok &&
```

Comment:
Should we still ignore update and delete here?

Reply:
Yes, update / delete should not be evolved since they would modify data, after this change, only Select would be considered by baseline evolution.